portafly/doc/ux_specs/global_components/modal.md

```diff
@@ -0,0 +1,52 @@
+    * Items are displayed in alphabetical order
+    * When the list of selected items is bigger than 5 items, only the first 5 will be displayed
+      * also, a button will be displayed at the bottom of the list to trigger display of the remaining items
+    * When the full list is displayed, the height of the container will be fixed and the content will be scrollable so that the entire dialog can still be seen above the fold
+  * Subject field > text input ([PF specs](https://www.patternfly.org/v4/documentation/react/components/textinput))
+    * mandatory field
+  * Message field > text area ([PF specs](https://www.patternfly.org/v4/documentation/react/components/textarea))
+    * mandatory field
+  * Buttons:
+    * Submit button [Disabled state until all conditions required to submit the form are met]
+    * Cancel button [Always enabled]
+      * Dismisses modal
+
+
+### _'Change plan for selected items'_ modal
+* Form ([PF specs](https://www.patternfly.org/v4/documentation/react/components/form))
+  * List of selected items > unordered list ([PF specs](https://www.patternfly.org/v4/documentation/react/components/list))
+    * Items are displayed in alphabetical order
+    * When the list of selected items is bigger than 5 items, only the first 5 will be displayed
+      * also, a button will be displayed at the bottom of the list to trigger display of the remaining items
+    * When the full list is displayed, the height of the container will be fixed and the content will be scrollable so that the entire dialog can still be seen above the fold
+  * Select plan field > dropdown ([PF specs](https://www.patternfly.org/v4/documentation/react/components/dropdown))
+    * mandatory field
+  * Buttons:
+    * Submit button [Disabled state until all conditions required to submit the form are met]
+    * Cancel button [Always enabled]
+      * Dismisses modal
+
+
+### _'Change state for selected items'_ modal
+* Form ([PF specs](https://www.patternfly.org/v4/documentation/react/components/form))
+  * List of selected items > unordered list ([PF specs](https://www.patternfly.org/v4/documentation/react/components/list))
+    * Items are displayed in alphabetical order
+    * When the list of selected items is bigger than 5 items, only the first 5 will be displayed
+      * also, a button will be displayed at the bottom of the list to trigger display of the remaining items
+    * When the full list is displayed, the height of the container will be fixed and the content will be scrollable so that the entire dialog can still be seen above the fold
+  * Select state field > dropdown ([PF specs](https://www.patternfly.org/v4/documentation/react/components/dropdown))
+    * mandatory field
+  * Buttons:
+    * Submit button [Disabled state until all conditions required to submit the form are met]
+    * Cancel button [Always enabled]
+      * Dismisses modal
```

Comment:
Should we also add stuff for keyboard support or is that baked in? Like. I expect 'esc' to dismiss a model as well.

Reply:
I thought that was implicit, but it's actually a good point. @damianpm @josemigallas wdyt?

Reply:
Well, I think it's always better to be explicit. A cancel button always being enabled (like in line 51) can be considered also implicit 🤷‍♂️
I think everything that we expect from the UI should be listed here. Anything not listed is optional.